### DISCUSSION TITLE
use pyvalhalla with a remote valhalla server

### DISCUSSION THREAD

**Discussion by Bondify** [gis-ops/pyvalhalla#33] [General]
hi, and first of all: thank you for such an amazing project :) 

I've been using Valhalla for some time now through HTTP request and would like to explore switching to `pyvalhalla` as it seems to be much faster (as per your comments in https://github.com/valhalla/valhalla/discussions/3796#discussioncomment-3922444)

Nonetheless, I would like to understand if I can call a remote Valhalla server with pyvalhalla or if it has to be deployed locally. From what I saw in the documentation, it sets the config from a locally generated graph `(config = get_config(tile_extract='./custom_files/valhalla_tiles.tar', verbose=True)
)`, but I'd need it to look for the tiles in the server where the valhalla image was deployed.

Thanks for your help,

**Comment by nilsnolde** [comment]
Jep, it has to be deployed locally, i.e. the graph has to be local to your calling code. Otherwise there's no benefit. The only thing this saves is the HTTP roundtrip, as I mentioned in that comment.

**Reply by Bondify** [reply]
I see, that makes sense. Thanks a lot for the quick reply @nilsnolde
